libraries/ESP8266mDNS/ESP8266mDNS.cpp

```diff
@@ -632,31 +649,45 @@ void MDNSResponder::_parsePacket(){
         uint16_t answerPrio = _conn_read16(); // Read priority
         uint16_t answerWeight = _conn_read16(); // Read weight
         answerPort = _conn_read16(); // Read port
+        last_bufferpos = 0;
 
         (void) answerPrio;
         (void) answerWeight;
 
         // Read hostname
         tmp8 = _conn_read8();
-        if (tmp8 & 0xC0) { // Compressed pointer (not supported)
+        if (tmp8 & 0xC0) { // Compressed pointer
+          uint16_t offset = ((((uint16_t)tmp8) & 0xC0) << 8) | _conn_read8();
```

Comment:
Do you think the calculated offset can be checked beeing out of bound to prevent crashing with malformed or crafted dns replies ?